src/Symfony/Component/Translation/Formatter/MessageFormatter.php

```diff
@@ -39,11 +39,7 @@ public function __construct(TranslatorInterface $translator = null, IntlFormatte
      */
     public function format(string $message, string $locale, array $parameters = [])
     {
-        if ($this->translator instanceof TranslatorInterface) {
-            return $this->translator->trans($message, $parameters, null, $locale);
-        }
-
-        return strtr($message, $parameters);
+        return $this->translator->trans($message, $parameters, null, $locale);
```

Comment:
`$this->translator` cannot be anything else than a `TranslatorInterface`, as typed in the constructor.